runtime/resource.go

```diff
@@ -0,0 +1,113 @@
+// Package runtime is a service runtime manager
+package runtime
+
+import "fmt"
+
+const (
+	TypeNamespace     = "namespace"
+	TypeNetworkPolicy = "networkpolicy"
+	TypeService       = "service"
+)
+
+// Resource represents any resource handled by runtime
+type Resource interface {
+	ID() string
+	Type() string
+}
+
+// Namespace represents a logical namespace for organising resources
+type Namespace struct {
+	// Name of the namespace
+	Name string
+}
+
+// NewNamespace mints a new namespace
+func NewNamespace(name string) (*Namespace, error) {
+	if name == "" {
+		return nil, ErrInvalidResource
+	}
+	return &Namespace{
+		Name: name,
+	}, nil
+}
+
+// ID implements Resource
+func (r *Namespace) ID() string {
+	return r.Name
+}
+
+// Type implements Resource
+func (*Namespace) Type() string {
+	return TypeNamespace
+}
+
+// NetworkPolicy represents an ACL of label pairs allowing ignress to a namespace
+type NetworkPolicy struct {
+	// The labels allowed ingress by this policy
+	AllowedLabels map[string]string
+	// Name of the network policy
+	Name string
+	// Namespace the network policy belongs to
```

Comment:
I think this should not be an attribute on the struct but be retrieved from the options as we currently do.